Dart/src/com/jetbrains/lang/dart/ide/errorTreeView/DartProblemsView.java

```diff
@@ -172,6 +181,44 @@ public static DartProblemsView getInstance(@NotNull final Project project) {
     return ServiceManager.getService(project, DartProblemsView.class);
   }
 
+  @SuppressWarnings("unused")
+  public void showWarningNotification(@NotNull Project project, @NotNull String title, @NotNull String htmlContent, @Nullable Icon icon) {
+    showNotification(NotificationType.WARNING, project, title, htmlContent, icon);
+  }
+
+  public void showErrorNotification(@NotNull Project project, @NotNull String title, @NotNull String htmlContent, @Nullable Icon icon) {
+    showNotification(NotificationType.ERROR, project, title, htmlContent, icon);
+  }
+
+  public void clearNotifications() {
+    if (myNotification != null) {
+      myNotification.expire();
+      myNotification = null;
+    }
+  }
+
+  private void showNotification(@NotNull NotificationType notificationType,
+                                @NotNull Project project,
```

Comment:
`Project` parameter seems to be redundant here and in 2 methods above, we have `myProject` field in hands. I'm on it, no actions needed.